stl/inc/mutex

```diff
@@ -624,14 +624,7 @@ public:
         static_assert(chrono::is_clock_v<_Clock>, "Clock type required");
 #endif // _HAS_CXX20
         while (!_Pred()) {
-            const auto _Now = _Clock::now();
-            if (_Abs_time <= _Now) {
-                return false;
-            }
-

```

Comment:
About this change:
1. This is specified by the standard and is in the most maintainable form. The correctness is now obviously only dependent on `wait_until(lock,abs_time,pred)`.
2. Actually this change fixes a bug - `if (_Abs_time <= _Now)` should return `_Pred()` too.
3. The efficiency loss is not significant IMO, and we will gain some efficiency back after fixing the clock-awareness bug.
4. `condition_variable_any` already does like this.

Reply:
(2) is not a bug. `return false` _is_ returning the result of the most-recent evaluation of `_Pred()`. (I do agree with the rest of this comment ;))